src/components/license-activation/LicenseActivationPage.jsx

```diff
@@ -1,13 +1,52 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { Redirect, useParams } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import { Alert } from '@edx/paragon';
+import { logInfo } from '@edx/frontend-platform/logging';
 
+import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+
+import { useRenderContactHelpText } from '../../utils/hooks';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
+import { LICENSE_STATUS } from '../enterprise-user-subsidy/data/constants';
+import LicenseActivationErrorAlert from './LicenseActivationErrorAlert';
 import LicenseActivation from './LicenseActivation';
-import AuthenticatedPage from '../app/AuthenticatedPage';
 
-const LicenseActivationPage = () => (
-  <AuthenticatedPage useEnterpriseConfigCache={false}>
-    <LicenseActivation />
-  </AuthenticatedPage>
+export const LOADING_MESSAGE = 'Your enterprise license is being activated! You will be automatically redirected to your organization\'s learner portal shortly.';
+
+const LicenseActivationPage = () => {
+  const user = getAuthenticatedUser();
+  const { enterpriseConfig } = useContext(AppContext);
+  const { subscriptionLicense } = useContext(UserSubsidyContext);
+  const { activationKey } = useParams();
+  const renderContactHelpText = useRenderContactHelpText(enterpriseConfig);
+
+  if (!subscriptionLicense || subscriptionLicense.status !== LICENSE_STATUS.ASSIGNED) {
+    logInfo(`User ${user.id} has no license to activate.`);
```

Comment:
nit/curious: I wonder if it would be helpful to separate this conditional into 2 separate `logInfo`, so we can log if learners simply don't have a license at all or if they have an incorrect license status. Alternatively, maybe include the `subscriptionLicense.uuid` in the existing `logInfo` if `subscriptionLicense` does indeed exist.